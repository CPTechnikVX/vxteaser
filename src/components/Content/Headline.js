import React      from 'react';
import PropTypes  from 'prop-types';
import Constants  from './../../utils/Constants';
import classnames from 'classnames';

export default class Headline extends React.PureComponent {
	render() {
		const {children, config, modifier, type} = this.props;
		const classList                          = [];

		classList.push(Constants.ClassName.Headline);

		if (type) {
			classList.push(Constants.PrefixClassName.Type + type);
		}

		if (modifier) {
			classList.push(modifier);
		}

		if (config && typeof config.aspectRatio !== 'undefined') {
			classList.push(Constants.ClassName.Headline + '--' + Constants.Element.Tile);
		}

		return <div className={classnames(classList)}>{children}</div>;
	}
}

Headline.propTypes = {
	/** @ignore */
	children: PropTypes.node,
	/** @ignore */
	config:   PropTypes.object,
	/**
	 * CSS modifier
	 */
	modifier: PropTypes.string,
	type:     PropTypes.oneOf(['main', 'sub', 'text']),
};

Headline.defaultProps = {
	type: 'text',
};
