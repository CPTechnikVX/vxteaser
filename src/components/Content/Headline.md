```js
<Headline type="main">
  <Text>VISIT-X Community Magazin</Text>
</Headline>
```
```js
<Headline type="sub">Januar 2018 - Ausgabe</Headline>
```

```js
<Headline>Just a paragrahp</Headline>
```
