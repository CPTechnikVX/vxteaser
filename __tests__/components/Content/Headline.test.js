import React     from 'react';
import {shallow} from 'enzyme';
import Headline  from '../../../src/components/Content/Headline';

describe('render headline', () => {
	test('empty headline', () => {
		const wrapper = shallow(
			<Headline>This is headline</Headline>
		);
		expect(wrapper).toMatchSnapshot();
	});
	test('main headline', () => {
		const wrapper = shallow(
			<Headline type="main">This is a main headline</Headline>
		);
		expect(wrapper).toMatchSnapshot();
	});
	test('sub headline', () => {
		const wrapper = shallow(
			<Headline type="sub">This is a sub</Headline>
		);
		expect(wrapper).toMatchSnapshot();
	});
	test('default text headline', () => {
		const wrapper = shallow(
			<Headline type="text">This is text</Headline>
		);
		expect(wrapper).toMatchSnapshot();
	});
});
