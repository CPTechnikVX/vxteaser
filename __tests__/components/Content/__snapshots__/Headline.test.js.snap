// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render headline default text headline 1`] = `
<div
  className="vxteaser-headline -type-text"
>
  This is text
</div>
`;

exports[`render headline empty headline 1`] = `
<div
  className="vxteaser-headline -type-text"
>
  This is headline
</div>
`;

exports[`render headline main headline 1`] = `
<div
  className="vxteaser-headline -type-main"
>
  This is a main headline
</div>
`;

exports[`render headline sub headline 1`] = `
<div
  className="vxteaser-headline -type-sub"
>
  This is a sub
</div>
`;
